Foundation/NSJSONSerialization.swift

```diff
@@ -271,19 +281,24 @@ private struct JSONWriter {
     
     mutating func serializeJSON(_ obj: Any) throws {
 
+        // For better performance, the most expensive conditions to evaluate should be last.
         switch (obj) {
         case let str as String:
             try serializeString(str)
         case let boolValue as Bool:
             serializeBool(boolValue)
-        case _ where _SwiftValue.store(obj) is NSNumber:
-            try serializeNumber(_SwiftValue.store(obj) as! NSNumber)
+        case let num as Int:
+            writer(String(describing: num))
```

Comment:
The current implementation represents the value as an `NSNumber` and then formats it using `CFNumberFormatterCreateStringWithNumber` using a `CFNumberFormatter` created by the `JSONSerialization` class.  It is the CF code (and format string that is passed to the formatter) determines how a number is represented.
However, this is significantly slower than using String.  Do you foresee a risk that the format of an Int / UInt produced by String may change?
An alternative could be to replicate what String is doing within `JSONWriter`. (is this what you had in mind?)